# Baseline Models

Three baselines, evaluated with strict **perturbation-level** train/test splits (see `src/data/splits.py`): a perturbation's cells never appear in both train and test, and two-gene combinations are further stratified by how many of their two genes were ever seen as single-gene perturbations during training (0/1/2 unseen).

- **No-perturbation**: always predict zero change.
- **Mean-shift**: always predict the average delta across training perturbations.
- **Ridge regression**: gene-pair embedding → expression delta, compared with two embedding variants (see `05_deep_learning.ipynb` for what the embeddings are).

In [1]:
import pandas as pd
pd.set_option('display.width', 200)
metrics = pd.read_csv('../results/tables/baseline_metrics.csv')
metrics.round(3)

,model,category,pearson_delta,mse,de_pearson_delta,direction_accuracy,n_perturbations
0,No-perturbation,test_combo_0_unseen,0.000,0.007,0.000,0.000,22
1,No-perturbation,test_combo_1_unseen,0.000,0.008,0.000,0.000,51
2,No-perturbation,test_combo_2_unseen,0.000,0.012,0.000,0.000,5
3,No-perturbation,test_unseen_single,0.000,0.003,0.000,0.000,26
4,No-perturbation,"train (seen, sanity check)",0.000,0.005,0.000,0.000,132
5,Mean-shift,test_combo_0_unseen,0.573,0.005,0.652,0.851,22
6,Mean-shift,test_combo_1_unseen,0.395,0.007,0.373,0.786,51
7,Mean-shift,test_combo_2_unseen,0.398,0.011,0.379,0.824,5
8,Mean-shift,test_unseen_single,0.337,0.003,0.336,0.762,26
9,Mean-shift,"train (seen, sanity check)",0.506,0.004,0.555,0.817,132


**Reading this table:** `pearson_delta` is the correlation between predicted and true expression change across all 2000 modeled genes; `de_pearson_delta` and `direction_accuracy` restrict to the top-50 genes by true |delta| (the genes that actually matter). `test_combo_2_unseen` has only 5 perturbations — treat it as a qualitative signal, not a precise estimate.

Even Ridge regression with a good gene-pair representation clears the mean-shift baseline by a wide margin on `test_combo_0_unseen` (0.65–0.90 vs. 0.57), confirming the embeddings carry real signal rather than the model just learning a global average response.